atlp-rwanda/rca-barfoot-team3-bn
# 184637995 - Send email verification code

#### What does this PR do?
Sending a verification code to user email after registration

#### Description of Task to be completed?
I have added
* nodemailer module to send emails
* updated our user model with new fields 
      1. verified
      2. verification_code
      3. verification_code_expiry_date

#### How should this be manually tested?
* Clone the app
* Setup mail configuration and database in .env
* run yarn dev
* hit a POST request on `/api/v1/users` to register the user
* check if you receive an email containing the verification code as well as the verification link
* hit a POST request on link that was embedded in the email `example: ` `/api/v1/users/verify/:email` send the code you received on the email.

#### What are the relevant pivotal tracker/Trello stories?
#184637995  - An Email should be sent to a user upon successful registration. The user should be able to receive a verification link via email upon successful signup

#### Screenshots (if appropriate)

<img width="1136" alt="Screenshot 2023-03-21 at 19 22 23" src="https://user-images.githubusercontent.com/50081035/226691341-001a1ed0-6c61-4817-b729-ffb05cd57e67.png">


<img width="1147" alt="Screenshot 2023-03-21 at 19 22 37" src="https://user-images.githubusercontent.com/50081035/226691471-c5ec255b-24f2-4b71-9b31-aa2f798b5e4f.png">
